activerecord/lib/active_record/connection_handling.rb

```diff
@@ -333,6 +347,14 @@ def clear_cache! # :nodoc:
       connection_pool.schema_cache.clear!
     end
 
+    def shard_keys
+      connection_class_for_self.instance_variable_get(:@shard_keys) || []
+    end
+
+    def sharded?
+      shard_keys.any?
```

Comment:
```suggestion
      shard_keys.length > 0
```
`#any?` takes an optional block and is slightly slower than e.g. `!shard_keys.empty?` or `shard_keys.length > 0`.

Reply:
I like the suggested change but I can see if others have thoughts.

Reply:
Is `any?` much slower? It reads better to me.
The contributing style guide doesn't mention anything about it and I see both style frequently in Rails's codebase.

Reply:
> Is `any?` much slower?

No, the difference is negligible. And I think we can afford to optimize for readability here.